src/crimson/osd/pg.cc

```diff
@@ -513,6 +513,38 @@ Context *PG::on_clean()
   return nullptr;
 }
 
+seastar::future<> PG::clear_temp_objects()
+{
+  logger().info("{} {}", __func__, pgid);
+  ghobject_t _next;
+  ceph::os::Transaction t;
+  uint64_t removed = 0;
+  auto max_size = local_conf()->osd_target_transaction_size;
+  while(true) {
+    auto [objs, next] = co_await shard_services.get_store().list_objects(
+      coll_ref, _next, ghobject_t::get_max(), max_size);
+    if (objs.empty()) {
+      if (!t.empty()) {
+        co_await shard_services.get_store().do_transaction(
+          coll_ref, std::move(t));
+      }
+      break;
+    }
+    for (auto &obj : objs) {
+      if (obj.hobj.is_temp()) {
+        t.remove(coll_ref->get_cid(), obj);
+        ++removed;
+      }
+    }
+    _next = next;
+    if (removed >= max_size) {
```

Comment:
nit: we can use `t.get_num_ops` instead of `removed`

Reply:
Done:-)